### PR TITLE
fix(HXIcon): correct issues with HXIconElement

### PR DESCRIPTION
**Pull request by CITguy**
* add documentation around `<hx-icon>` being an inline element
* add Shadow Markup for HXIconElement
* add Shadow Styles for HXIconElement
* remove unnecessary `@import` for hx-icon styles
* remove unnecessary `fill-rule` attributes from SVG files

Deprecates: 
* `HXIconElement.icons` in favor of `HelixUI.ICONS` (See #309)

Closes:
* SURF-1224 (inconsistent button heights w/ icons)
* SURF-1260 (multiple rendering of hx-icon)
* SURF-1293 (JSX: `<hx-error>`)
* SURF-1272 (JSX: Icons)
* SURF-1114 - fix focus styles of "X" in `<hx-search>`

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 3c8606b9024a089d72fdca7c99f15368d589ac30

https://app.netlify.com/sites/helix-ui/deploys/5b722a2ac9659216a01414fe

File: src/helix-ui/elements/HXPillElement.less [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
-  display: none;
-  height: 1.5em;
-  margin: 0 -0.5rem 0 0;
-  width: 1.5em;
-  
-  hx-icon {
-    font-size: 0.75em;
+    display: flex;
+    height: 100%;
+    width: 100%;
```
Comment (100stacks) [inline src/helix-ui/elements/HXPillElement.less]: move these up and alphabetize

File: src/helix-ui/elements/HXPillElement.less [hunk 2]
```diff
@@ -1,39 +1,41 @@
...
+    &:hover {
+      color: @gray-1000;
+    }
   }
 }
 
 :host([dismissable]) {
   #hxDismiss {
-    display: inline-block;
+    display: inline-flex;;
```
Comment (100stacks) [inline src/helix-ui/elements/HXPillElement.less]: please remove extra `;`

**Comment by palindromebart** [comment]
design LGTM

File: src/helix-ui/elements/HXIconElement.less [hunk 3]
```diff
@@ -0,0 +1,26 @@
+:host {
+  background-color: transparent;
+  color: currentColor;
+  display: inline-block;
+  flex-shrink: 0;
+  height: 1em;
+  width: 1em;
+
+  #hxIcon {
+    background-color: transparent;
```
Comment (cathysiller) [inline src/helix-ui/elements/HXIconElement.less]: check if needed

File: src/helix-ui/elements/HXIconElement.less [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    box-sizing: border-box;
+    display: block;
+    flex-shrink: 0;
+    height: 100%;
+    line-height: 1;
+    vertical-align: baseline;
```
Comment (cathysiller) [inline src/helix-ui/elements/HXIconElement.less]: double check if needed

**Pull request merged**